### PR TITLE
Improve and expose app bundle generator

### PR DESCRIPTION
**Pull request by jmcphers**
Today there are two parallel implementations of the Shiny app bundle generator -- that is, the function that returns the list of files that should be bundled, given a directory. There's one in the IDE and one here in the **rsconnect** package. They do roughly the same thing but have slightly different features; for instance, the one in the IDE halts recursion after reaching a size cap, but ignores all dotfiles, whereas the one in this package has an unbounded size but has a more specific list of dotfiles it ignores. 

This has generated some confusion, so this change merges the behaviors and moves the unified function to the package (a later change to the ID will invoke the package function to generate the bundle file list).

The net change (from the package perspective) is:

1. There is a new option, `rsconnect.max.bundle.size`. If an attempt is made to deploy a bundle exceeding this size, an error is generated.
2. A new function is exposed which generates the bundle file list. This can be used by a front end (i.e. the IDE) to generate the file list.

### CODE REVIEW

File: R/onLoad.R [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.onLoad <- function(libname, pkgname) {
+  if (is.null(getOption("rsconnect.max.bundle.size"))) {
+    options(rsconnect.max.bundle.size = 1048576000)
```
Comment (jspiewak) [inline R/onLoad.R]: The current plans for shinyapps.io allow:

```
            'free': 1000000000,  # 1GB
            'starter': 1000000000,  # 1GB
            'basic': 2000000000,  # 2GB
            'standard': 50000000000,  # 5GB
            'professional': 50000000000,  # 5GB
```

Comment (jmcphers) [inline R/onLoad.R]: Thanks! Is this exposed anywhere? I don't think the package today knows anything about service plans.

Comment (jspiewak) [inline R/onLoad.R]: For a given user, it is possible to obtain their BundleSizeLimit via an API call.

Comment (jmcphers) [inline R/onLoad.R]: Does/will that exist on Connect as well? Or is that purely a shinyapps.io construct?

Comment (jspiewak) [inline R/onLoad.R]: Great question that I defer to the @rstudio/connect team.

Comment (jmcphers) [inline R/onLoad.R]: Also: does that size limit apply to the bundle (compressed) or the app itself (uncompressed)?

Comment (jspiewak) [inline R/onLoad.R]: It is the size limit of the payload, i.e. we check the `Content-Length` header.

Practically speaking the 5 GB limit is the threshold at which you cease being able to upload to S3 in a single PUT call.

We probably should switch to multi-part for everything, per their recommendation. But really what I would like to do is supply `rsconnect` with a signed S3 URL and not have shinyapps.io in the middle at all.

https://docs.aws.amazon.com/AmazonS3/latest/dev/UploadingObjects.html

Comment (aronatkins) [inline R/onLoad.R]: Connect does not have bundle limits.

**Comment by jjallaire** [comment]
Merged so we can get this into tonight's daily build. Still TBD is what to do about enforcing the correct limit for shinyapps.io

**Pull request merged**